regulations/static/regulations/js/source/views/main/child-view.js

```diff
@@ -104,20 +105,23 @@ var ChildView = Backbone.View.extend({
                 if (_.isEmpty(this.activeSection) || (this.activeSection !== this.$sections[i].id)) {
                     this.activeSection = this.$sections[i][0].id;
                     this.$activeSection = $(this.$sections[i][0]);
-                    // **Event** trigger active section change
-                    HeaderEvents.trigger('section:open', this.activeSection);
-                    MainEvents.trigger('paragraph:active', this.activeSection);
-
-                    if (typeof window.history !== 'undefined' && typeof window.history.replaceState !== 'undefined') {
-                        // update hash in url
-                        window.history.replaceState(
-                            null,
-                            null,
-                            window.location.origin + window.location.pathname + window.location.search + '#' + this.activeSection
-                        );
+                    if (this.$activeSection.is(':visible')) {
```

Comment:
Sorry, let me explain what I'm seeing. The major context I have is that there was a chunk of code which was always being called and now it's only being called if something is visible. I interpreted that as "we hit a situation we can't handle, so put a flag around it". Trying to dive deeper, I'm guessing the scroll position when in write mode shouldn't affect the ToC (as the DOM is completely different). Is that what's going on?

If we know that `PreambleView` is exceptional here, due to having a write mode which messes with the scroll position, yes, I'd think overriding a function (hence confining the logic to the exceptional path) would be a good strategy.